pr-deploy/test/app.test.ts

```diff
@@ -0,0 +1,116 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// mock unzipAndMove dependency before importing '../src/app'
+jest.mock('../src/zipper', () => {
+  return {
+    unzipAndMove: jest.fn().mockReturnValue(Promise.resolve('gs://serving-bucket/prId')),
+  };
+});
+
+import {Application} from 'probot';
+import {prDeployAppFn} from '../src/app';
+import express from 'express';
+import request from 'supertest';
+import Webhooks, {
+  WebhookPayloadPullRequest,
+  WebhookPayloadCheckRun} from '@octokit/webhooks';
+
+describe('test pr deploy app', () => {
+  let app: Application;
+  let github: any;
+  let server: express.Application;
+
+  beforeEach(() => {
+    github = {
+      checks: {
+        create: jest.fn().mockReturnValue(Promise.resolve()),
+        update: jest.fn().mockReturnValue(Promise.resolve()),
+        listForRef: jest.fn().mockImplementation(
+          params => {
+            if (params.repo == 'newRepo') {
+              return null;
+            } else {
+              return {data: {total_count: 1, check_runs: [{id: 12345}]}};
+            }
+          }),
+      },
+    };
+
+    app = new Application();
+    app.load(prDeployAppFn);
+    app.auth = async() => Promise.resolve(github);
+
+    server = express();
+    server.use(app.router);
+  });
+
+  test('creates a check when a pull request is opened', async() => {
+    const prOpenedEvent: Webhooks.WebhookEvent<WebhookPayloadPullRequest> = {
+      id: 'prId',
+      name: 'pull_request.opened',
+      payload: {
+        pull_request: {head: {sha: 'abcde'}},
+        repository: {owner: {name: 'repoOwner'},name: 'newRepo'},
+      } as WebhookPayloadPullRequest,
+    };
+
+    await app.receive(prOpenedEvent);
+    expect(github.checks.create).toHaveBeenCalled();
+    expect(github.checks.update).not.toHaveBeenCalled();
+  });
+
+  test('refreshes the check when a pull request is synchronized or reopened',
+    async() => {
+      const prSynchronizedEvent:
+      Webhooks.WebhookEvent<WebhookPayloadPullRequest> = {
+        id: 'prId',
+        name: 'pull_request.synchronize',
+        payload: {
+          pull_request: {head: {sha: 'abcde'}},
+          repository: {owner: {name: 'repoOwner'},name: 'existingRepo'},
+        } as WebhookPayloadPullRequest,
+      };
+
+      await app.receive(prSynchronizedEvent);
+      expect(github.checks.update).toHaveBeenCalled();
+      expect(github.checks.create).not.toHaveBeenCalled();
+    });
+
+  test('enables deployment action when post is received', async done => {
+    request(server)
+      .post('/pr-deploy/owners/1/repos/2/headshas/3')
+      .expect(() => { expect(github.checks.update).toHaveBeenCalledTimes(1);})
```

Comment:
Why not have that as a separate line after the `request();`? 🤔

Reply:
Yeah it's not intuitive! The assertion fails if you do it after the request. It's something with how  `request()` from `supertest` works.